Runtime/Components/Cinemachine3rdPersonFollow.cs

```diff
@@ -139,56 +158,103 @@ void PositionCamera(ref CameraState curState, float deltaTime)
             var followTargetForward = followTargetRotation * fwd;
             var angle = UnityVectorExtensions.SignedAngle(
                 fwd, followTargetForward.ProjectOntoPlane(up), up);
-            var previousHeadingAngle = deltaTime >= 0 ? PreviousHeadingAngle : angle;
+            var previousHeadingAngle = deltaTime >= 0 ? m_PreviousHeadingAngle : angle;
             var deltaHeading = angle - previousHeadingAngle;
-            PreviousHeadingAngle = angle;
+            m_PreviousHeadingAngle = angle;
 
             // Bypass user-sourced rotation
-            dampedTargetPos = targetPos 
-                + Quaternion.AngleAxis(deltaHeading, up) * (dampedTargetPos - targetPos);
-            PreviousFollowTargetPosition = dampedTargetPos;
+            dampedTargetPos = followTarget 
+                + Quaternion.AngleAxis(deltaHeading, up) * (dampedTargetPos - followTarget);
+            m_PreviousFollowTargetPosition = dampedTargetPos;
 
-            GetRigPositions(out Vector3 root, out Vector3 shoulder, out Vector3 hand);
+            GetRigPositions(out Vector3 root, out _, out Vector3 hand);
 
             // 1. Check if pivot itself is colliding with something, if yes, then move the pivot 
             // closer to the player. The radius is bigger here than in step 2, to avoid problems 
             // next to walls. Where the preferred distance would be pulled completely to the 
             // player, using a bigger radius, this won't happen.
-            hand = PullTowardsStartOnCollision(in root, in hand, in CameraCollisionFilter, CameraRadius * 1.05f);
+            bool handHasHit = PullTowardsStartOnCollision(in root, in hand, in CameraCollisionFilter, 
+                CameraRadius * 1.05f, out var handResolved);
+            // Post collision damping
+            Vector3 dampedHandResolved = DampedPullBackPostCollision(handHasHit, deltaTime, root, handResolved, 
+                ref m_PrevHandHasHit, ref m_IsHandPostCollision, ref m_PrevHandDistance);
 
             // 2. Try to place the camera to the preferred distance
-            var camPos = hand - (followTargetForward * CameraDistance);
-            camPos = PullTowardsStartOnCollision(in hand, in camPos, in CameraCollisionFilter, CameraRadius);
+            Vector3 camPos = dampedHandResolved - (followTargetForward * CameraDistance);
+            bool camPosHasHit = PullTowardsStartOnCollision(in dampedHandResolved, in camPos, in CameraCollisionFilter, 
+                CameraRadius, out var camPosResolved);
+            // Post collision damping
+            Vector3 dampedCamPosResolved = DampedPullBackPostCollision(camPosHasHit, deltaTime, root, camPosResolved, 
+                ref m_PrevCamPosHasHit, ref m_IsCamPosPostCollision, ref m_PrevCamPosDistance);
```

Comment:
you have introduced a lot of extra variables here, not sure why it's needed.  It's confusing.